### ISSUE TITLE
problems when there are top level filesystem mount points 

### ISSUE THREAD

**Issue by mccahill** [binpash/try#38]
In the code that tries to mount the overlays, there is a check for top level directory mount points that is a little too aggressive for systems the have user filesystems mounted at locations like /srv and /home (which is the custom of the system administrators where I work).

this line

           if [ -d "$top_dir_abs" ] && ! mountpoint -q "$top_dir_abs"; then
          
excludes the LVM volumes mounted at /srv/and /home on my system via the "! mountpoint" clause. 
When I run try from my home directory (/home/mccahill) it fails like this:
  
            /tmp/tmp.P5nVCaSZ5L: line 4: cd: /home/mccahill: No such file or directory
            
 Try does run correctly when run from a directory for which it has created an overlay. 
 I can see why try wants to exclude mounts like /boot, /dev, /proc, /run, /sys, and so on, but user filesystem mounts need to have overlays.

**Comment by ericzty** [comment]
Thanks for the report! Could you please try the branch [mount-fix](https://github.com/binpash/try/tree/mount-fix) and let me know if it works on your setup.

**Comment by mtolly** [comment]
I encountered this same issue, and yes the `mount-fix` branch fixes it for me :raised_hands:

**Comment by skorokithakis** [comment]
I have the same issue, and the current `main` does not fix it for me:

```
16:06:01 $ try pip3 install libdash
Warning: Failed mounting /home as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /tmp as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /usr as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /run/user/1000/doc as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /home/stavros as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /usr/local as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/games as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/log as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib/AccountsService as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib/apt as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib/NetworkManager as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib/dpkg as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib/docker/zfs/graph/ab70723c0d03ace2dab092988bdb726b9c7da165ef903741ed6d85d7a7489ca4 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/lib/docker/zfs/graph/c2276dcb514032531472e4f20341225fe99c0cdad1af3b87812b2789481ea9de as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/snap as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/snap/firefox/common/host-hunspell as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/mail as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/www as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /var/spool as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /boot as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /boot/efi as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /boot/grub as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/bare/5 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/core20/1891 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/core20/1950 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/core22/750 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/core22/766 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/firefox/2710 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/firefox/2760 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/gnome-3-38-2004/137 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/gnome-3-38-2004/140 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/gnome-42-2204/105 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/gnome-42-2204/111 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/gtk-common-themes/1535 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/snap-store/638 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/snap-store/959 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/snapd/19361 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/snapd/19457 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/snapd-desktop-integration/57 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /snap/snapd-desktop-integration/83 as an overlay, see /tmp/tmp.R14M4UGHNQ
Warning: Failed mounting /tmp/.mount_desktoh3wGH2 as an overlay, see /tmp/tmp.R14M4UGHNQ
/tmp/tmp.0wS2EKl09A: line 4: cd: /tmp/try: No such file or directory
```

**Comment by mgree** [comment]
@skorokithakis I wonder if your problem is #19. Can you show us the output of `findmnt -A` and `lsblk`?

**Comment by skorokithakis** [comment]
Sure:

```
16:47:23 $ findmnt -A
TARGET                                                                               SOURCE                               FSTYPE                      OPTIONS
/                                                                                    rpool/ROOT/ubuntu_0rbgsw             zfs                         rw,relatime,xattr,posixacl
├─/sys                                                                               sysfs                                sysfs                       rw,nosuid,nodev,noexec,relatime
│ ├─/sys/kernel/security                                                             securityfs                           securityfs                  rw,nosuid,nodev,noexec,relatime
│ ├─/sys/fs/cgroup                                                                   cgroup2                              cgroup2                     rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot
│ ├─/sys/fs/pstore                                                                   pstore                               pstore                      rw,nosuid,nodev,noexec,relatime
│ ├─/sys/firmware/efi/efivars                                                        efivarfs                             efivarfs                    rw,nosuid,nodev,noexec,relatime
│ ├─/sys/fs/bpf                                                                      bpf                                  bpf                         rw,nosuid,nodev,noexec,relatime,mode=700
│ ├─/sys/kernel/debug                                                                debugfs                              debugfs                     rw,nosuid,nodev,noexec,relatime
│ ├─/sys/kernel/tracing                                                              tracefs                              tracefs                     rw,nosuid,nodev,noexec,relatime
│ ├─/sys/fs/fuse/connections                                                         fusectl                              fusectl                     rw,nosuid,nodev,noexec,relatime
│ └─/sys/kernel/config                                                               configfs                             configfs                    rw,nosuid,nodev,noexec,relatime
├─/proc                                                                              proc                                 proc                        rw,nosuid,nodev,noexec,relatime
│ └─/proc/sys/fs/binfmt_misc                                                         systemd-1                            autofs                      rw,relatime,fd=29,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=23252
│   └─/proc/sys/fs/binfmt_misc                                                       binfmt_misc                          binfmt_misc                 rw,nosuid,nodev,noexec,relatime
├─/dev                                                                               udev                                 devtmpfs                    rw,nosuid,relatime,size=7951664k,nr_inodes=1987916,mode=755,inode64
│ ├─/dev/pts                                                                         devpts                               devpts                      rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000
│ ├─/dev/shm                                                                         tmpfs                                tmpfs                       rw,nosuid,nodev,inode64
│ ├─/dev/hugepages                                                                   hugetlbfs                            hugetlbfs                   rw,relatime,pagesize=2M
│ └─/dev/mqueue                                                                      mqueue                               mqueue                      rw,nosuid,nodev,noexec,relatime
├─/run                                                                               tmpfs                                tmpfs                       rw,nosuid,nodev,noexec,relatime,size=1605968k,mode=755,inode64
│ ├─/run/lock                                                                        tmpfs                                tmpfs                       rw,nosuid,nodev,noexec,relatime,size=5120k,inode64
│ ├─/run/credentials/systemd-sysusers.service                                        ramfs                                ramfs                       ro,nosuid,nodev,noexec,relatime,mode=700
│ ├─/run/credentials/systemd-tmpfiles-setup-dev.service                              ramfs                                ramfs                       ro,nosuid,nodev,noexec,relatime,mode=700
│ ├─/run/credentials/systemd-sysctl.service                                          ramfs                                ramfs                       ro,nosuid,nodev,noexec,relatime,mode=700
│ ├─/run/credentials/systemd-tmpfiles-setup.service                                  ramfs                                ramfs                       ro,nosuid,nodev,noexec,relatime,mode=700
│ ├─/run/snapd/ns                                                                    tmpfs[/snapd/ns]                     tmpfs                       rw,nosuid,nodev,noexec,relatime,size=1605968k,mode=755,inode64
│ │ ├─/run/snapd/ns/firefox.mnt                                                      nsfs[mnt:[4026533041]]               nsfs                        rw
│ │ └─/run/snapd/ns/snapd-desktop-integration.mnt                                    nsfs[mnt:[4026533175]]               nsfs                        rw
│ ├─/run/user/1000                                                                   tmpfs                                tmpfs                       rw,nosuid,nodev,relatime,size=1605964k,nr_inodes=401491,mode=700,uid=1000,gid=1000,inode64
│ │ ├─/run/user/1000/gvfs                                                            gvfsd-fuse                           fuse.gvfsd-fuse             rw,nosuid,nodev,relatime,user_id=1000,group_id=1000
│ │ └─/run/user/1000/doc                                                             portal                               fuse.portal                 rw,nosuid,nodev,relatime,user_id=1000,group_id=1000
│ ├─/run/docker/netns/2ead400bd7fc                                                   nsfs[net:[4026533211]]               nsfs                        rw
│ └─/run/docker/netns/72095a979a5e                                                   nsfs[net:[4026533195]]               nsfs                        rw
├─/home/stavros                                                                      rpool/USERDATA/stavros_vnfof6        zfs                         rw,relatime,xattr,posixacl
├─/root                                                                              rpool/USERDATA/root_vnfof6           zfs                         rw,relatime,xattr,posixacl
├─/srv                                                                               rpool/ROOT/ubuntu_0rbgsw/srv         zfs                         rw,relatime,xattr,posixacl
├─/var/lib                                                                           rpool/ROOT/ubuntu_0rbgsw/var/lib     zfs                         rw,relatime,xattr,posixacl
│ ├─/var/lib/AccountsService                                                         rpool/ROOT/ubuntu_0rbgsw/var/lib/AccountsService
│ │                                                                                                                       zfs                         rw,relatime,xattr,posixacl
│ ├─/var/lib/NetworkManager                                                          rpool/ROOT/ubuntu_0rbgsw/var/lib/NetworkManager
│ │                                                                                                                       zfs                         rw,relatime,xattr,posixacl
│ ├─/var/lib/apt                                                                     rpool/ROOT/ubuntu_0rbgsw/var/lib/apt zfs                         rw,relatime,xattr,posixacl
│ ├─/var/lib/dpkg                                                                    rpool/ROOT/ubuntu_0rbgsw/var/lib/dpkg
│ │                                                                                                                       zfs                         rw,relatime,xattr,posixacl
│ ├─/var/lib/docker/zfs/graph/ab70723c0d03ace2dab092988bdb726b9c7da165ef903741ed6d85d7a7489ca4
│ │                                                                                  rpool/ROOT/ubuntu_0rbgsw/var/lib/ab70723c0d03ace2dab092988bdb726b9c7da165ef903741ed6d85d7a7489ca4
│ │                                                                                                                       zfs                         rw,relatime,xattr,posixacl
│ └─/var/lib/docker/zfs/graph/c2276dcb514032531472e4f20341225fe99c0cdad1af3b87812b2789481ea9de
│                                                                                    rpool/ROOT/ubuntu_0rbgsw/var/lib/c2276dcb514032531472e4f20341225fe99c0cdad1af3b87812b2789481ea9de
│                                                                                                                         zfs                         rw,relatime,xattr,posixacl
├─/var/games                                                                         rpool/ROOT/ubuntu_0rbgsw/var/games   zfs                         rw,relatime,xattr,posixacl
├─/usr/local                                                                         rpool/ROOT/ubuntu_0rbgsw/usr/local   zfs                         rw,relatime,xattr,posixacl
├─/var/log                                                                           rpool/ROOT/ubuntu_0rbgsw/var/log     zfs                         rw,relatime,xattr,posixacl
├─/var/snap                                                                          rpool/ROOT/ubuntu_0rbgsw/var/snap    zfs                         rw,relatime,xattr,posixacl
│ └─/var/snap/firefox/common/host-hunspell                                           rpool/ROOT/ubuntu_0rbgsw[/usr/share/hunspell]
│                                                                                                                         zfs                         ro,noexec,noatime,xattr,posixacl
├─/var/mail                                                                          rpool/ROOT/ubuntu_0rbgsw/var/mail    zfs                         rw,relatime,xattr,posixacl
├─/var/www                                                                           rpool/ROOT/ubuntu_0rbgsw/var/www     zfs                         rw,relatime,xattr,posixacl
├─/var/spool                                                                         rpool/ROOT/ubuntu_0rbgsw/var/spool   zfs                         rw,relatime,xattr,posixacl
├─/boot                                                                              bpool/BOOT/ubuntu_0rbgsw             zfs                         rw,nodev,relatime,xattr,posixacl
│ ├─/boot/efi                                                                        /dev/nvme0n1p1                       vfat                        rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
│ └─/boot/grub                                                                       /dev/nvme0n1p1[/grub]                vfat                        rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro
├─/snap/bare/5                                                                       /dev/loop0                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/core20/1891                                                                  /dev/loop1                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/core20/1950                                                                  /dev/loop2                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/core22/750                                                                   /dev/loop3                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/core22/766                                                                   /dev/loop4                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/firefox/2710                                                                 /dev/loop5                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/firefox/2760                                                                 /dev/loop6                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/gnome-3-38-2004/137                                                          /dev/loop7                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/gnome-3-38-2004/140                                                          /dev/loop8                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/gnome-42-2204/105                                                            /dev/loop9                           squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/gnome-42-2204/111                                                            /dev/loop10                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/gtk-common-themes/1535                                                       /dev/loop11                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/snap-store/638                                                               /dev/loop12                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/snap-store/959                                                               /dev/loop13                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/snapd/19361                                                                  /dev/loop14                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/snapd/19457                                                                  /dev/loop15                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/snapd-desktop-integration/57                                                 /dev/loop16                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
├─/snap/snapd-desktop-integration/83                                                 /dev/loop17                          squashfs                    ro,nodev,relatime,errors=continue,threads=single
└─/tmp/.mount_desktoysblPB                                                           desktop-3.6.0.AppImage               fuse.desktop-3.6.0.AppImage ro,nosuid,nodev,relatime,user_id=1000,group_id=1000

16:47:27 $ lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0         7:0    0     4K  1 loop /snap/bare/5
loop1         7:1    0  63,5M  1 loop /snap/core20/1891
loop2         7:2    0  63,4M  1 loop /snap/core20/1950
loop3         7:3    0  73,8M  1 loop /snap/core22/750
loop4         7:4    0  73,9M  1 loop /snap/core22/766
loop5         7:5    0 242,9M  1 loop /snap/firefox/2710
loop6         7:6    0 244,8M  1 loop /snap/firefox/2760
loop7         7:7    0 349,7M  1 loop /snap/gnome-3-38-2004/137
loop8         7:8    0 349,7M  1 loop /snap/gnome-3-38-2004/140
loop9         7:9    0 460,7M  1 loop /snap/gnome-42-2204/105
loop10        7:10   0 466,5M  1 loop /snap/gnome-42-2204/111
loop11        7:11   0  91,7M  1 loop /snap/gtk-common-themes/1535
loop12        7:12   0  45,9M  1 loop /snap/snap-store/638
loop13        7:13   0  12,3M  1 loop /snap/snap-store/959
loop14        7:14   0  53,3M  1 loop /snap/snapd/19361
loop15        7:15   0  53,3M  1 loop /snap/snapd/19457
loop16        7:16   0   428K  1 loop /snap/snapd-desktop-integration/57
loop17        7:17   0   452K  1 loop /snap/snapd-desktop-integration/83
nvme0n1     259:0    0 931,5G  0 disk 
├─nvme0n1p1 259:1    0   512M  0 part /boot/grub
│                                     /boot/efi
├─nvme0n1p2 259:2    0     2G  0 part [SWAP]
├─nvme0n1p3 259:3    0     2G  0 part 
└─nvme0n1p4 259:4    0   927G  0 part 
```

**Comment by mgree** [comment]
Okay, I think your issue is indeed #19.

**Comment by ericzty** [comment]
I agree, does it work if you unmount `/tmp/.mount_desktoysblPB` ? @skorokithakis

**Comment by skorokithakis** [comment]
It does not:

```
20:17:24 $ sudo umount -f /tmp/.mount_deskto3rbIlV
umount: /tmp/.mount_deskto3rbIlV: target is busy.

20:17:34 $ sudo umount -fl /tmp/.mount_deskto3rbIlV

stavros@talos:~ (master|✚4) 
20:17:46 $ try pip3 install libdash
Warning: Failed mounting /home as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /usr as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /run/user/1000/doc as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /home/stavros as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib/AccountsService as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib/NetworkManager as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib/apt as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib/dpkg as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib/docker/zfs/graph/ab70723c0d03ace2dab092988bdb726b9c7da165ef903741ed6d85d7a7489ca4 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/lib/docker/zfs/graph/c2276dcb514032531472e4f20341225fe99c0cdad1af3b87812b2789481ea9de as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/games as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /usr/local as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/log as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/mail as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/snap as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/snap/firefox/common/host-hunspell as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/spool as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /var/www as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /boot as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /boot/efi as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /boot/grub as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/bare/5 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/core20/1891 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/core20/1950 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/core22/750 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/core22/766 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/firefox/2710 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/firefox/2760 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/gnome-3-38-2004/137 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/gnome-3-38-2004/140 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/gnome-42-2204/105 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/gnome-42-2204/111 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/gtk-common-themes/1535 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/snap-store/638 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/snap-store/959 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/snapd/19361 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/snapd/19457 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/snapd-desktop-integration/57 as an overlay, see /tmp/tmp.CLH0bJRRum
Warning: Failed mounting /snap/snapd-desktop-integration/83 as an overlay, see /tmp/tmp.CLH0bJRRum
/tmp/tmp.HKiE92wdmR: line 4: cd: /home/stavros: No such file or directory
```

**Comment by ericzty** [comment]
Old error: `/tmp/tmp.0wS2EKl09A: line 4: cd: /tmp/try: No such file or directory`
New error: `/tmp/tmp.HKiE92wdmR: line 4: cd: /home/stavros: No such file or directory`
What happens if you try running try in temp again? Now it is failing because /home/stavros is nested.

Also, could you try with branch nested-mount (need to install mergerfs) (and running from your home dir)
If you don't like mergerfs, you can install unionfs-fuse and change the two appearances of mergerfs to unionfs (L70, L83)

**Comment by angelhof** [comment]
Should be addressed by #67
